### PR TITLE
fix noble hash keccak256 usage

### PR DESCRIPTION
**Pull request by cb-jake**
### _Summary_

This PR https://github.com/coinbase/coinbase-wallet-sdk/pull/1336 removed sha.js and keccak/js in in favor of @noble/hashes but implemented the keccak256 hash incorrectly. This resulted in EIP712 messages failing for walletlink.

This fixes the change by using the functions how theyre intended to be used.

### _How did you test your changes?_

1. Create a local build (`cd ./packages/wallet-sdk && rm -rf dist && yarn build && cd ../../`) 
2. Start the dev server from root (`yarn dev`)
3. Open up the playground `localhost:3000`
4. Connect to the playground using walletlink
5. Test signed typed data v3 and v4 and ensure you're able to sign on mobile (Note v1 doesn't work on sdk v4.0.4 either)

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/vendor-js/eth-eip712-util/index.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/* eslint-disable */
+//prettier-ignore
+
```
Comment (nateReiners) [inline packages/wallet-sdk/src/vendor-js/eth-eip712-util/index.js:3]: why is this needed now but wasn't before if there are no other changes in this file? 🤔

Comment (cb-jake) [reply packages/wallet-sdk/src/vendor-js/eth-eip712-util/index.js:3]: I was logging in this file to trace down the issue and lint was trying to update everything. To prevent that moving forward it made sense to ignore this file as well

**Review by nateReiners — DISMISSED** [review]

**Review by vishnumad — DISMISSED** [review]

**Review by nateReiners — APPROVED** [review]
lgtm! 

tested the following locally: 

eth_signTypedData_v3 with walletlink
eth_signTypedData_v4 with walletlink
eth_signTypedData_v3 with extension
eth_signTypedData_v4 with extension
eth_signTypedData_v3 with smart wallet 
eth_signTypedData_v4 with smart wallet
